### PR TITLE
E2E: Support windows arm32

### PR DESCRIPTION
**Pull request by and-rewsmith**
We are currently initializing the iotedge installation with agent image 1.0 on startup. This won't work for windows arm32 (or any future new platform under development) so I am adding support for a quickstart arg "initialize-with-agent-artifact" that can be used to initialize the installation with the desired agent artifact, skipping 1.0.

### CODE REVIEW

**Comment by and-rewsmith** [comment]
Successful Win arm32 test: 
https://msazure.visualstudio.com/One/_releaseProgress?_a=release-environment-logs&releaseId=1741313&environmentId=4613930

File: smoke/IotEdgeQuickstart/Quickstart.cs [hunk 1]
```diff
@@ -34,16 +35,18 @@ public class Quickstart : Details.Details
...
             Option<DPSAttestation> dpsAttestation)
-            : base(bootstrapper, credentials, iothubConnectionString, eventhubCompatibleEndpointWithEntityPath, upstreamProtocol, proxy, imageTag, deviceId, hostname, deploymentFileName, twinTestFileName, deviceCaCert, deviceCaPk, deviceCaCerts, optimizedForPerformance, runtimeLogLevel, cleanUpExistingDeviceOnSuccess, dpsAttestation)
+            : base(bootstrapper, credentials, iothubConnectionString, eventhubCompatibleEndpointWithEntityPath, upstreamProtocol, proxy, imageTag, deviceId, hostname, deploymentFileName, twinTestFileName, deviceCaCert, deviceCaPk, deviceCaCerts, optimizedForPerformance, initializeWithAgentArtifact, runtimeLogLevel, cleanUpExistingDeviceOnSuccess, dpsAttestation)
         {
             this.leaveRunning = leaveRunning;
             this.noVerify = noVerify;
             this.bypassEdgeInstallation = bypassEdgeInstallation;
             this.verifyDataFromModule = verifyDataFromModule;
             this.dpsProvisionTest = dpsAttestation.HasValue;
+            this.initializeWithAgentArtifact = initializeWithAgentArtifact;
```
Comment (philipktlin) [inline smoke/IotEdgeQuickstart/Quickstart.cs]: why do you need this? it is already passed to base class.

File: smoke/IotEdgeQuickstart/details/IotedgedLinux.cs [hunk 2]
```diff
@@ -174,9 +174,17 @@ public Task Install()
                 300); // 5 min timeout because install can be slow on raspberry pi
         }
 
-        public async Task Configure(DeviceProvisioningMethod method, string hostname, string deviceCaCert, string deviceCaPk, string deviceCaCerts, LogLevel runtimeLogLevel)
+        public async Task Configure(DeviceProvisioningMethod method, Option<string> agentImage, string hostname, string deviceCaCert, string deviceCaPk, string deviceCaCerts, LogLevel runtimeLogLevel)
         {
-            Console.WriteLine($"Setting up iotedged with agent image 1.0");
+            agentImage.ForEach(
```
Comment (philipktlin) [inline smoke/IotEdgeQuickstart/details/IotedgedLinux.cs]: is it better to combine this ForEach with ForEach on line 232?

Comment (and-rewsmith) [reply smoke/IotEdgeQuickstart/details/IotedgedLinux.cs]: I thought about this too, but liked this executing at the very beginning.

File: smoke/IotEdgeQuickstart/Quickstart.cs (hunk 1 above)
Comment (and-rewsmith) [reply smoke/IotEdgeQuickstart/Quickstart.cs]: Good catch, I don't

File: smoke/IotEdgeQuickstart/details/IotedgedLinux.cs (hunk 2 above)
Comment (philipktlin) [reply smoke/IotEdgeQuickstart/details/IotedgedLinux.cs]: nit: should put together; will make sure any updates will not miss the other?  your call.

Comment (and-rewsmith) [reply smoke/IotEdgeQuickstart/details/IotedgedLinux.cs]: My only concern is the unhappy path where something fails. If we move the log statement down then we will see less information in this error case.

**Pull request merged**